frb_rust/src/support.rs

```diff
@@ -50,3 +50,52 @@ pub struct WireSyncReturnStruct {
     pub len: i32,
     pub success: bool,
 }
+
+impl From<WireSyncReturnData> for WireSyncReturnStruct {
+    fn from(data: WireSyncReturnData) -> Self {
+        let (ptr, len) = into_leak_vec_ptr(data.0);
+        WireSyncReturnStruct {
+            ptr,
+            len,
+            success: true,
+        }
+    }
+}
+
+/// Wrapper struct for [`WireSyncReturnStruct`].
+pub struct WireSyncReturnData(Vec<u8>);
+
+impl From<Vec<u8>> for WireSyncReturnData {
+    fn from(data: Vec<u8>) -> Self {
+        WireSyncReturnData(data)
+    }
+}
+
+/// Bool will be converted to u8 where 0 stands for false and 1 stands for true.
+impl From<bool> for WireSyncReturnData {
+    fn from(data: bool) -> Self {
+        if data { 1_u8 } else { 0_u8 }.to_be_bytes().to_vec().into()
```

Comment:
uber-nit: We have implemented `u8` into `WireSyncReturnData`. So no need to `.to_be_bytes().to_vec()`

Reply:
whoops